crates/sui-indexer/src/indexer_reader.rs

```diff
@@ -1030,7 +1030,8 @@ impl IndexerReader {
                     )
                 }
                 EventFilter::MoveEventType(struct_tag) => {
-                    format!("event_type = '{}'", struct_tag)
+                    let formatted_struct_tag = struct_tag.to_canonical_string(true);
+                    format!("event_type = '{formatted_struct_tag}'")
```

Comment:
Just a small tweak here -- we can use `to_canonical_display` to avoid materializing to an intermediate string.

```suggestion
                    format!(
                        "event_type = '{}'",
                        struct_tag.to_canonical_display(/* with_prefix */ true),
                    )
```

Reply:
Hey @amnn thanks for your response and suggestion, will keep this in mind! 🙏